<?xml version="1.0"?>
<launch>
    <!-- 加载参数配置 -->
    <param name="multicast_addr" value="224.0.1.2"/>
    <param name="multicast_port" value="9999"/>
    <param name="car_name_file" value="$(env HOME)/.config/autofleet/car_name.txt"/>

    <!-- 加载机器人模型描述 -->
    <param name="robot_description" 
           command="$(find xacro)/xacro '$(find car_monitor)/urdf/racecar.xacro'"/>

    <!-- 启动 robot_state_publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
        <param name="publish_frequency" type="double" value="50.0"/>
    </node>

    <!-- 启动 joint_state_publisher -->
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher">
        <param name="use_gui" value="false"/>
    </node>

    <!-- 设置坐标转换 -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_laser" 
          args="0.093603 0 0.12377 0 0 0 base_link lidar"/>
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_camera"
          args="0.14851 0.0022137 0.0975 0 0 0 base_link camera"/>

    <!-- 启动通信节点 -->
    <node name="car_monitor" pkg="car_monitor" type="car_monitor_node" output="screen">
        <rosparam command="load" file="$(find car_monitor)/config/car_monitor.yaml"/>
    </node>

    <!-- 启动 RViz -->
    <node name="rviz" pkg="rviz" type="rviz" 
          args="-d $(find car_monitor)/rviz_config/monitor.rviz"/>

</launch>

